src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -840,6 +857,21 @@ internal async Task InvokeResponse(InvocationResponse invokeResponse)
                         SendCloseSharedMemoryResourcesForInvocationRequest(outputMaps);
                     }
                 }
+
+                string data = $"WorkerId: {Id}, TotalInvocations : {_invocationMetricsPerWorkerId[Id].TotalInvocations}, SuccessfulInvocations : {_invocationMetricsPerWorkerId[Id].SuccessfulInvocations}, AverageInvocationLatency : {_invocationMetricsPerWorkerId[Id].AverageInvocationLatency}";
+                _metricsLogger.LogEvent(string.Format(MetricEventNames.WorkerInvocationStatus, Id), data: data);
+            }
+
+            if (_invocationTimer == null)
```

Comment:
This is not thread safe - two concurrent invocations may both initialize a timer. This will lead to double callbacks of `OnInvocationTimer` and also a failure to dispose one of them. Can we initialize the timer in the ctor, then just start it here?

Reply:
Done